### PR TITLE
DOC: documentation for ZMQ API for communication with RE Manager.

### PR DESCRIPTION
**Pull request by dmgav**
First version of ZMQ API for communicating with RE Manager. The API can be used by third party applications for communicating to RE Manager. It is also used by other modules of the QueueServer, including `qserver` CLI tool and HTTP web server.

Related issue: https://github.com/bluesky/bluesky-queueserver/issues/106

### CODE REVIEW

File: docs/source/re_manager_api.rst [hunk 1]
```diff
@@ -0,0 +1,809 @@
...
+
+**'manager_stop'**
+^^^^^^^^^^^^^^^^^^
+
+
+============  =========================================================================================
+Method        **'manager_stop'**
+------------  -----------------------------------------------------------------------------------------
+Description   Exit RE Manager application. Clients will probably not need to initiate exit remotely,
+              but ability to do so it is extremely useful for automated testing.
```
Comment (mrakitin) [inline docs/source/re_manager_api.rst:769]: "... to do so it is extremely ..." --> "... to do so is extremely ..."

**Review by mrakitin — COMMENTED** [review]
I did my late, after-merge review, and spotted a few issues in the documentation. I hope it can be implemented in one of the future PRs.

File: docs/source/re_manager_api.rst [hunk 2]
```diff
@@ -0,0 +1,809 @@
...
+- :ref:`method_queue_start`
+- :ref:`method_queue_stop`
+- :ref:`method_queue_stop_cancel`
+
+Send commands to Bluesky Run Engine:
+
+- :ref:`method_re_pause`
+- :ref:`method_re_resume_stop_abort_halt`
+
+Stopping RE Manager (mostly used in testing):
```
[the remark this answers is not in the document]
Comment (dmgav) [reply docs/source/re_manager_api.rst:108]: Changes implemented in PR https://github.com/bluesky/bluesky-queueserver/pull/118

File: docs/source/re_manager_api.rst [hunk 3]
```diff
@@ -0,0 +1,809 @@
...
+                  indicates if the request was processed successfully.
+
+              **msg**: *str*
+                  error message in case of failure, empty string ('') otherwise.
+------------  -----------------------------------------------------------------------------------------
+Execution     The request only initiates the sequence of creating a new environment. It may take
+              a long time to start the environment. Monitor 'manager_state'
+              and 'worker_environment_exists' status fields ('status' method) to detect when
+              the operation is completed: 'manager_state' should have the value 'opening_environment'
+              while operation is in process and change to 'idle' when the operation completes and
```
[the remark this answers is not in the document]
Comment (dmgav) [reply docs/source/re_manager_api.rst:334]: Changes implemented in PR https://github.com/bluesky/bluesky-queueserver/pull/118

File: docs/source/re_manager_api.rst [hunk 4]
```diff
@@ -0,0 +1,809 @@
...
+              'worker_environment_exists' is set True if environment was created successfully.
```
[the remark this answers is not in the document]
Comment (dmgav) [reply docs/source/re_manager_api.rst:335]: Changes implemented in PR https://github.com/bluesky/bluesky-queueserver/pull/118

File: docs/source/re_manager_api.rst [hunk 5]
```diff
@@ -0,0 +1,809 @@
...
+.. _method_environment_destroy:
+
+**'environment_destroy'**
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+============  =========================================================================================
+Method        **'environment_destroy'**
+------------  -----------------------------------------------------------------------------------------
+Description   Initiate the operation of destroying of (unresponsive) the existing RE Worker
```
[the remark this answers is not in the document]
Comment (dmgav) [reply docs/source/re_manager_api.rst:378]: Changes implemented in PR https://github.com/bluesky/bluesky-queueserver/pull/118

File: docs/source/re_manager_api.rst [hunk 6]
```diff
@@ -0,0 +1,809 @@
...
+                  indicates if the request was processed successfully.
+
+              **msg**: *str*
+                  error message in case of failure, empty string ('') otherwise.
+------------  -----------------------------------------------------------------------------------------
+Execution     The request initiates the sequence of closing the environment.
+              Monitor 'manager_state' and 'worker_environment_exists' status fields
+              (see 'status' method) to detection when the operation coompletes:
+              'manager_state' is expected to have the value 'closing_environment'
+              while operation is in process and switch to 'idle' when the operation completes
```
[the remark this answers is not in the document]
Comment (dmgav) [reply docs/source/re_manager_api.rst:364]: Changes implemented in PR https://github.com/bluesky/bluesky-queueserver/pull/118

File: docs/source/re_manager_api.rst [hunk 7]
```diff
@@ -0,0 +1,809 @@
...
+------------  -----------------------------------------------------------------------------------------
+Returns       **success**: *boolean*
+                  indicates if the request was processed successfully.
+
+              **msg**: *str*
+                  error message in case of failure, empty string ('') otherwise.
+
+              **qsize**: *int* or *None*
+                  the number of items in the plan queue after the plan was added if
+                  the operation is successful, *None* otherwise
```
[the remark this answers is not in the document]
Comment (dmgav) [reply docs/source/re_manager_api.rst:468]: Changes implemented in PR https://github.com/bluesky/bluesky-queueserver/pull/118

File: docs/source/re_manager_api.rst [hunk 8]
```diff
@@ -0,0 +1,809 @@
...
+
+
+============  =========================================================================================
+Method        **'queue_item_remove'**
+------------  -----------------------------------------------------------------------------------------
+Description   Remove item from the queue. By default the last item in the queue is removed.
+              Alternatively the position or UID of the item can be specified.
+------------  -----------------------------------------------------------------------------------------
+Parameters    **pos**: *int*, *'front'* or *'back'* (optional)
+                  position of the item in the queue. RE Manager will attempt to insert the item
```
[the remark this answers is not in the document]
Comment (dmgav) [reply docs/source/re_manager_api.rst:524]: Changes implemented in PR https://github.com/bluesky/bluesky-queueserver/pull/118

File: docs/source/re_manager_api.rst [hunk 9]
```diff
@@ -0,0 +1,809 @@
...
+                  at the specified position. The position may be positive or negative (counted
+                  from the back of the queue) integer. If 'pos' value is a string 'front' or 'back',
+                  then the item is inserted at the front or the back of the queue.
```
[the remark this answers is not in the document]
Comment (dmgav) [reply docs/source/re_manager_api.rst:527]: Changes implemented in PR https://github.com/bluesky/bluesky-queueserver/pull/118

File: docs/source/re_manager_api.rst [hunk 10]
```diff
@@ -0,0 +1,809 @@
...
+
+              **msg**: *str*
+                  error message in case of failure, empty string ('') otherwise.
+
+              **item**: *dict*
+                  the dictionary of parameters of the removed item, ({}) if operation failed.
+
+              **qsize**: *int* or *None*
+                  the number of items in the plan queue after the plan was added if
+                  the operation is successful, *None* otherwise
```
[the remark this answers is not in the document]
Comment (dmgav) [reply docs/source/re_manager_api.rst:545]: Changes implemented in PR https://github.com/bluesky/bluesky-queueserver/pull/118

File: docs/source/re_manager_api.rst (hunk 1 above)
Comment (dmgav) [reply docs/source/re_manager_api.rst:769]: Changes implemented in PR https://github.com/bluesky/bluesky-queueserver/pull/118

**Pull request merged**